File: Practica2/busquedalineal.py
import tracemalloc
import time


sizeArr = int(input("Ingresa el tamaño del arreglo que deseas: ")) + 1
numeroBuscar = int(input("Ingresa el numero que deseas buscar: "))

arr = list(range(1, sizeArr))

start = time.time()
tracemalloc.start()

def busquedaLineal(arr, numeroBuscar):
    posicionNumero = -1
    for idxb, numero in enumerate(arr):
        if(numero == numeroBuscar):
            posicionNumero = idxb
            return posicionNumero
    return posicionNumero


posicion = busquedaLineal(arr, numeroBuscar)
    
if(posicion != -1):
    print(f"Numero encontrado en la posicion {posicion}")
else:
    print(f"Numero no encontrado")



current, peak = tracemalloc.get_traced_memory()
print(f"Memoria actual: {current / 1024} KB; Pico: {peak / 1024} KB")
tracemalloc.stop()
end = time.time()
print(f"Tiempo de ejecución: {end - start} segundos")

File: Practica2/Busqueda Lineal.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int busquedaLineal(int arr[], int n, int x) {
    for (int i = 0; i < n; i++) {
        if (arr[i] == x) return i;
    }
    return -1;
}

int main() {
    int n, x;
    printf("Ingrese el tamaño del arreglo: ");
    scanf("%d", &n);

    int *arr = (int*)malloc(n * sizeof(int));
    if (arr == NULL) return 1;

    for (int i = 0; i < n; i++) arr[i] = i + 1;

    printf("Ingrese el valor a buscar: ");
    scanf("%d", &x);

    clock_t inicio = clock();
    int resultado = busquedaLineal(arr, n, x);
    clock_t fin = clock();

    double tiempo = (double)(fin - inicio) / CLOCKS_PER_SEC;
    size_t memoria = n * sizeof(int);

    if (resultado == -1)
        printf("No encontrado\n");
    else
        printf("Encontrado en la posición %d\n", resultado);

    printf("Tiempo: %f segundos\n", tiempo);
    printf("Memoria usada: %zu bytes\n", memoria);

    free(arr);
    return 0;
}